llvm/lib/Transforms/Scalar/CorrelatedValuePropagation.cpp

```diff
@@ -1075,20 +1076,50 @@ static bool processSExt(SExtInst *SDI, LazyValueInfo *LVI) {
   return true;
 }
 
-static bool processZExt(ZExtInst *ZExt, LazyValueInfo *LVI) {
-  if (ZExt->getType()->isVectorTy())
+static bool processPossibleNonNeg(Instruction *I, LazyValueInfo *LVI) {
+  assert(isa<PossiblyNonNegInst>(I) && "Requires PossiblyNonNeg instruction");
```

Comment:
Can you use `PossiblyNonNegInst *I` as param type instead? Or is the hierarchy not right for that currently?